test/org/zaproxy/zap/extension/pscanrulesBeta/ViewStateScannerTest.java

```diff
@@ -0,0 +1,197 @@
+package org.zaproxy.zap.extension.pscanrulesBeta;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertThat;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.junit.Before;
+import org.junit.Test;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.extension.encoder.Base64;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+public class ViewStateScannerTest extends PassiveScannerTest {
+	
+	private HttpMessage msg;
+	private HttpRequestHeader header;
+
+	@Before
+	public void before() throws URIException {
+		msg = new HttpMessage();
+		header = new HttpRequestHeader();
+		
+		msg.setRequestHeader(header);
+		header.setURI(new URI("http://does.not.matter.com", true));
+	}
+	
+	@Override
+	protected PluginPassiveScanner createScanner() {
+		return new ViewstateScanner();
+	}
+	
+	@Test
+	public void shouldPassAsThereIsNoContent() {
```

Comment:
Perhaps "shouldNotRaiseAlertAsThereIsNoContent" or something like that....obviously tests "shouldPass". Same for others below.